### DISCUSSION TITLE
Empty domain parts

### DISCUSSION THREAD

**Discussion by kfryd** [mtex-toolbox/mtex#1397] [Ask Anything] [answered]
Hello,

I would like to plot the EBSD-like maps based on CPFEM results.

My problem is that the domain boundary is not in the shape of a rectangle:
![image](https://user-images.githubusercontent.com/10358809/162416651-30e7a295-546d-429c-b800-17120e6d1614.png)

So when I'm calculating the grains I am not happy with the result:
![image](https://user-images.githubusercontent.com/10358809/162416806-96ad37c9-b495-4940-90e9-1960029c4e0d.png)

Do you know how I can fix it?

Kind regards,
Karol Frydrych

**Comment by kilir** [comment, marked as answer]
Hi Karol,
can you try  `calcGrains(....'boundary', 'tight')`, eventually followed by anumber somewhat <0.95 which I think is  the current default.
Cheers,
Rüdiger

**Reply by kfryd** [reply]
Thank you, it seems to do the job.

**Reply by kfryd** [reply]
And do you know what can I do if I have an empty space (say, disk or ellipse) surrounded by the material? tight does not work in this case...

**Reply by kilir** [reply]
Any chance to assign `'notIndexed'` to that inclusion?

**Reply by kilir** [reply]
There's also the option `'maxDistance', distance_in_meas_units ` which limits the distance between two points to be within one grain. However in the case of e.g. a thin elliptical inclusion, this may not work.

**Reply by kfryd** [reply]
Thank you Rudiger for your suggestions. Indeed, it was very easy to add a set of points on the boundary of the inscribed ellipse with another phase which does the job very nicely.
